packages/web-frontend/src/tests/historyNavigation/UrlPeriodStringToDateRangeConversions.test.js

```diff
@@ -0,0 +1,56 @@
+import moment from 'moment';
+import {
+  convertUrlPeriodStringToDateRange,
+  convertDateRangeToUrlPeriodString,
+} from '../../historyNavigation/utils';
+
+const SINGLE_PERIOD_STRING = moment('2020-01-02').format('Do_MMM_YYYY');
+const SINGLE_DATE_RANGE = {
+  startDate: moment('2020-01-02'),
+  endDate: moment('2020-01-02'),
+};
+
+const RANGE_PERIOD_STRING = `${moment('2020-01-02').format('Do_MMM_YYYY')}-${moment(
+  '2020-01-10',
+).format('Do_MMM_YYYY')}`;
+
+const RANGE_DATE_RANGE = {
+  startDate: moment('2020-01-02'),
+  endDate: moment('2020-01-10'),
+};
+
+describe('convertUrlPeriodStringToDateRange', () => {
+  it('should decode strings with only one date', () => {
+    const { startDate, endDate } = convertUrlPeriodStringToDateRange(
+      SINGLE_PERIOD_STRING,
+      'one_day_at_a_time',
+    );
+    const { startDate: expectedStartDate, endDate: expectedEndDate } = SINGLE_DATE_RANGE;
+    expect(startDate.isSame(expectedStartDate)).toBe(true);
```

Comment:
Using `isSame` is clever and clean! 💯

Can I suggest an optional change: it took me a bit to realise that `isSame` comes from `moment` and not from jest (although at a closer look, the syntax does not imply that). You can add a `assertMomentDatesAreEqual ` helper:

```js
const assertMomentDatesAreEqual = (moment1, moment2) => expect(moment1.isSame(moment2)).toBe(true)`
```

Reply:
Clean!